### PR TITLE
UI changes for Gig editing

### PR DESCRIPTION
**Pull request by bcongdon**
Addresses https://github.com/acm-uiuc/groot-gigs-service/issues/2

### CODE REVIEW

File: app/controllers/gigs.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ module.exports = function(app) {
         title: req.body.title,
         details: req.body.details,
         credits: parseInt(req.body.credits),
-        admin_task: req.body.admin_gig
+        admin_task: req.body.admin_gig ? true : false
```
Comment (sameetandpotatoes) [inline app/controllers/gigs.js]: Why was this line changed from the original?

File: app/controllers/gigs.js [hunk 2]
```diff
@@ -176,12 +176,45 @@ module.exports = function(app) {
...
+  app.post('/intranet/gigs/edit/:gig_id', function(req, res) {
+    request({
+      url: `${SERVICES_URL}/gigs/${req.params.gig_id}`,
+      method: "GET",
+      headers: {
+        "Authorization": GROOT_ACCESS_TOKEN,
+      },
+      json: true
+    }, function(err, response, body) {
+      if(body.issuer != req.session.student.netid) {
```
Comment (sameetandpotatoes) [inline app/controllers/gigs.js]: May want to implement that default error handling that we tend to do for requests i.e 

```
if (err || !response || body.statusCode != 200) {

}
```

Honestly, it may be worth considering to make these request blocks a helper function (not in this PR, just in general) that has this error check already so we can avoid a lot of code duplication and make our actual controller logic a bit shorter.

Comment (bcongdon) [inline app/controllers/gigs.js]: Yeah, I've thought that too. There is a lot of code duplication, and most of it never gets tested...

File: app/controllers/gigs.js (hunk 1 above)
Comment (bcongdon) [inline app/controllers/gigs.js]: I think this might not be needed anymore, there was an issue parsing the boolean value on the service side, but I think that was resolved.

Comment (bcongdon) [inline app/controllers/gigs.js]: Checked again, something like this is needed because of the way forms are submitted. I changed it to `Boolean(req.body.admin_gig)` to be a little bit more clear what this is doing.
